spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java

```diff
@@ -0,0 +1,257 @@
+package com.novoda.spritz;
+
+import android.animation.Animator;
+import android.animation.AnimatorSet;
+import android.animation.TimeInterpolator;
+import android.animation.ValueAnimator;
+import android.support.v4.view.ViewPager;
+import android.view.animation.LinearInterpolator;
+
+import com.airbnb.lottie.LottieAnimationView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class Spritz {
+
+    private final LottieAnimationView lottieAnimationView;
+    private final List<SpritzStepWithOffset> spritzStepsWithOffset;
+    private final long totalAnimationDuration;
+    private final long defaultSwipeAnimationDuration;
+    private final TimeInterpolator defaultSwipeForwardInterpolator;
+    private final TimeInterpolator defaultSwipeBackwardsInterpolator;
+
+    private ViewPager viewPager;
+    private float currentPosition;
+    private AnimatorSet animatorSet;
+
+    public static Builder with(LottieAnimationView lottieAnimationView) {
+        return new Builder(lottieAnimationView);
+    }
+
+    private Spritz(LottieAnimationView lottieAnimationView,
+                   List<SpritzStepWithOffset> spritzStepsWithOffset,
+                   long totalAnimationDuration,
+                   long defaultSwipeAnimationDuration,
+                   TimeInterpolator defaultSwipeForwardInterpolator,
+                   TimeInterpolator defaultSwipeBackwardsInterpolator) {
+
+        this.lottieAnimationView = lottieAnimationView;
+        this.spritzStepsWithOffset = spritzStepsWithOffset;
+        this.totalAnimationDuration = totalAnimationDuration;
+        this.defaultSwipeAnimationDuration = defaultSwipeAnimationDuration;
+        this.defaultSwipeForwardInterpolator = defaultSwipeForwardInterpolator;
+        this.defaultSwipeBackwardsInterpolator = defaultSwipeBackwardsInterpolator;
+        this.animatorSet = new AnimatorSet();
+    }
+
+    private void attach(ViewPager viewPager) {
+        this.viewPager = viewPager;
+
+        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
```

Comment:
The only real _business_ logic is in this method. Perhaps you can extract stuff and unit test it?